Atomic/delete.py

```diff
@@ -0,0 +1,61 @@
+from . import Atomic
+from . import util
+import sys
+import subprocess
+
+
+class Delete(Atomic):
+    def __init__(self):
+        super(Delete, self).__init__()
+
+    def delete_image(self):
+        """
+        Mark given image(s) for deletion from registry
+        :return: True if all images marked for deletion
+        """
+
+        if not self.args.force_delete:
+            confirm = util.input("Do you wish to delete {}? (y/N) ".format(self.args.delete_targets))
+            confirm = confirm.strip().lower()
+            if not confirm in ['y', 'yes']:
+                sys.stderr.write("User aborted delete operation for {}\n".format(self.args.delete_targets))
+                sys.exit(2)
+
+        if self.args.remote_delete:
+            results = self._delete_remote(self.args.delete_targets)
+        else:
+            results = self._delete_local(self.args.delete_targets)
+        return results
+
+    def _delete_remote(self, targets):
+        results = 0
+        for target in targets:
+            # _convert_to_skopeo requires registry v1 support while delete requires v2 support
+            # args, img = self.syscontainers._convert_to_skopeo(target)
+
+            args = []
+            if "http:" in target:
+                args.append("--insecure")
+
+            for i in ["oci:", "http:", "https:"]:
+                img = target.replace(i, "docker:")
+
+            if not img.startswith("docker:"):
+                img = "docker://" + img
+
+            try:
+                util.skopeo_delete(img, args)
+                sys.stdout.write("Image {} marked for deletion\n".format(img))
+            except ValueError as e:
+                sys.stderr.write("Failed to mark Image {} for deletion: {}\n".format(img, e))
+                results = 2
+        return results
+
+    def _delete_local(self, targets):
+        results = True
+        for target in targets:
+            try:
+                util.check_call([self.docker_binary(), "rmi", target])
+            except subprocess.CalledProcessError:
```

Comment:
Can you use the docker python bindings.

self.d.remove_image(target)

Also I think we might want to add support for SystemContainers, although I am not sure how you remove these images.

@giuseppe PTAL